machine-learning/_posts/2018-10-18-UsefulThingsML.md

```diff
@@ -0,0 +1,193 @@
+
+![](/machine-learning/images/UsefulThingsToKnowML/MLMethodComponents.png)
+
+
+2. **IT'S GENERALIZATION THAT COUNTS**
+
+The fundamental goal of machine learning is to **generalize beyond the
+examples in the training set**. This is because, no matter how much data we
+have, it is very unlikely that we will see those exact examples again at
+test time.
+
+The algorithm must perform well in unseen data; having a good training
+accuracy may mean that the method has just memorized the examples it has seen.
+
+**So keep testing data aside** for the final evaluation.
+
+
+3. **DATA ALONE IS NOT ENOUGH**
+
+Generalization being the goal has another major consequence: data alone is not
+enough, no matter how much of it you have. Learners combine knowledge with
+data to grow programs.
+
+
+4. **OVERFITTING HAS MANY FACES**
+
+If the knowledge and data we have are not sufficient to completely determine
+the correct classifier we run the risk of just hallucinating a classifier (or
+parts of it) that is not grounded in reality, and is simply encoding random
+quirks in the data.
+
+Overfitting comes in many forms that are not immediately obvious. One way to
+understand overfitting is by decomposing generalization error into bias and
+variance. Bias is a learner's tendency to consistently learn the same wrong
+thing. Variance is the tendency to learn random things irrespective of the
+real signal.
+
+* Strong false assumptions can be better than weak true ones, because a
+learner with the latter needs more data to avoid overfitting.
+* **Regularization** functions added to the evaluation function help ensuring
+that the method explains the data with the simplest possible model.
+* **Cross-validation** can help to mitigate overfitting.
+* A common misconception about overfitting is that it is caused by noise.
+* Control the fraction of falsely accepted non-null hypotheses, known as the
+_false discovery rate_.
+
+
+5. **INTUITION FAILS IN HIGH DIMENSIONS**
+
+The _curse of dimensionality_ refers to either the fact that methods that work
+fine in low dimensions become intractable when the input is high-dimensional,
+or to the fact that Generalizing correctly becomes exponentially harder as
+the dimensionality (number of features) of the examples grows.
+
+Our intuitions, which come from a threedimensional world, often do not apply
+in high-dimensional ones.
+
+
+6. **THEORETICAL GUARANTEES ARE NOT WHAT THEY SEEM**the
+
+_Machine learning papers are full of theoretical guarantees_.
+
+The main role of theoretical guarantees in machine learning is not as a
+criterion for practical decisions, but as a source of understanding and
+driving force for algorithm desig
+
+Just because a learner has a theoretical justification and works in practice
+practice doesn't mean the former is the reason for the latter.
+
+
+7. **FEATURE ENGINEERING IS THE KEY**
+
+At the end of the day, some machine learning projects succeed and some fail.

```

Comment:
I suggest adding a note before this paragraph (or somewhere else), to tell that feature engineering is becoming less important with deep learning. For example, with RGB image, engineered features are not used anymore.